solidity/contracts/converter/BancorFormula.sol

```diff
@@ -183,7 +183,9 @@ contract BancorFormula is IBancorFormula, Utils {
     */
     function purchaseRate(uint256 _supply, uint256 _reserveBalance, uint32 _reserveWeight, uint256 _depositAmount) public view returns (uint256) {

```

Comment:
@yudilevi:

Recall what I told you about revert messages in read-only functions:
- Read [this GitHub issue](https://github.com/ethereum/web3.js/issues/3435) that I have opened a while back
- Try calling `func(0)` in [this contract](https://ropsten.etherscan.io/address/0x13566a62a0aa3d4be5aae979a47cb2f69048977b#readContract) that I have deployed on Ropsten and verified on Etherscan a while back

In this contract (`BancorFormula`), all functions are read-only.
We will need to carefully verify that there are no implications of these revert messages here:
1. A non-constant function which calls any of these functions with invalid input should correctly revert
2. A user (dapp) which calls any of these functions with invalid input should get a revert-indication

I am more concerned about the 2nd bullet above.
From my experience, when you call such functions, you get an actual value instead of a revert-indication.
This will be catastrophic for users, who will try to execute transactions based on that (even if the transaction reverts, it will still cost gas).

Reply:
looking into the potential compiler/EVM issue

Reply:
@yudilevi:
A compiler/EVM issue may well be interpreted as a bug in our contracts.
But please note that even if it's not, it is still a dapp issue for sure (for example, a dapp might allow some conversion even though it is expected to revert).
If the problem is indeed web3.js, and its latest version indeed resolves the problem, then we will have to indicate that in our documentation, and perhaps even encourage our known users to upgrade their web3.js versions.